ionos_create_machine.go

```diff
@@ -0,0 +1,448 @@
+package ionoscloud
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/log"
+	"github.com/ionos-cloud/docker-machine-driver/internal/pointer"
+	sdkgo "github.com/ionos-cloud/sdk-go/v6"
+)
+
+func (d *Driver) getCubeTemplateUuid() (string, error) {
+	templates, err := d.client().GetTemplates()
+	if err != nil {
+		return "", err
+	}
+
+	for _, template := range *templates.Items {
+		if *template.Properties.Name == d.Template {
+			return *template.Id, nil
+		}
+	}
+	return "", err
+}
+
+func (d *Driver) addSSHUserToYaml() (string, error) {
+	commonUser := map[interface{}]interface{}{
+		"name":                d.SSHUser,
+		"lock_passwd":         true,
+		"sudo":                "ALL=(ALL) NOPASSWD:ALL",
+		"create_groups":       false,
+		"no_user_group":       true,
+		"ssh_authorized_keys": []string{d.SSHKey},
+	}
+
+	return d.client().UpdateCloudInitFile(d.CloudInit, "users", []interface{}{commonUser}, false, "append")
+}
+
+func (d *Driver) CreateDataCenterIfNeeded() (err error) {
+	var dc *sdkgo.Datacenter
+	if d.DatacenterId == "" {
+		d.DCExists = false
+		var err error
+		log.Debugf("Creating datacenter...")
+		dc, err = d.client().CreateDatacenter(d.DatacenterName, d.Location)
+		if err != nil {
+			return fmt.Errorf("error creating datacenter: %w", err)
+		}
+	} else {
+		d.DCExists = true
+		log.Debugf("Getting existing datacenter..")
+		dc, err = d.client().GetDatacenter(d.DatacenterId)
+		if err != nil {
+			return fmt.Errorf("error getting datacenter: %w", err)
+		}
+	}
+	if dcId, ok := dc.GetIdOk(); ok && dcId != nil {
+		d.DatacenterId = *dcId
+		log.Debugf("Datacenter ID: %v", d.DatacenterId)
+	}
+	return nil
+}
+
+func (d *Driver) CreateLanIfNeeded() (err error) {
+	if d.LanId == "" {
```

Comment:
(just a nitpick)

It would be easier to read if this is an early return

```suggestion
	if d.LanId != "" {
	    return nil
	}
	/// rest of the func not nested
```